src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -91,11 +91,7 @@ protected void MapDirectives(Action handler, params string[] directives)
         {
             foreach (var directive in directives)
             {
-                _directiveParsers.Add(directive, () =>
-                {
-                    EnsureDirectiveIsAtStartOfLine();
```

Comment:
Had to move this into each of the directive parsing bits so that the start of line errors made their way onto the `DirectiveIRNode`s and not the document.